plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/ui/SnapToAllDragEditPartsTracker.java

```diff
@@ -86,24 +128,145 @@ protected void snapPoint(ChangeBoundsRequest request) {
             snapToAllShape = !SnapToAllDragEditPartsTracker.DEFAULT_SNAP_TO_SHAPE_MODE;
-            return true;
+        } else if (!acceptArrowKey(event)) {
+            result = super.handleKeyDown(event);
+        } else {
+            moveWithArrowKeysSiriusMode = true;
+            // Enable the move behavior based on a request set according to the arrow direction pressed.
+
+            // With this mode, it is not possible to enable the SnapToAllDragEditPartsTracker.SNAP_TO_ALL mode. Indeed,
+            // this mode is enabled by pressing the F4 key during a mouse move. It was also not available in the classic
+            // GMF/GEF mode.
+
+            // By the way, the previous mode does not correctly handle the snap to shape (but it is currently not the
+            // goal of this new mode to fix this).
+
+            // Reproduce the "acceleration behavior" of
+            // org.eclipse.gef.tools.DragEditPartsTracker.handleKeyDown(KeyEvent) to get the increment to use.
+            siriusAccStepIncrement();
+            PrecisionPoint moveDelta;
+            switch (event.keyCode) {
+            case SWT.ARROW_DOWN:
+                moveDelta = new PrecisionPoint(0, siriusAccGetStep());
+                break;
+            case SWT.ARROW_UP:
+                moveDelta = new PrecisionPoint(0, -siriusAccGetStep());
+                break;
+            case SWT.ARROW_RIGHT:
+                int stepping = siriusAccGetStep();
+                if (isCurrentViewerMirrored2()) {
+                    stepping = -stepping;
+                }
+                moveDelta = new PrecisionPoint(stepping, 0);
+                break;
+            case SWT.ARROW_LEFT:
+                int step = -siriusAccGetStep();
+                if (isCurrentViewerMirrored2()) {
+                    step = -step;
+                }
+                moveDelta = new PrecisionPoint(step, 0);
+                break;
+            default:
+                moveDelta = new PrecisionPoint(0, 0);
+            }
+            // Adapt the moveDelta to the zoom level, because later in
+            // org.eclipse.gef.editpolicies.ConstrainedLayoutEditPolicy.getConstraintFor(ChangeBoundsRequest,
+            // GraphicalEditPart) the compute is done with coordinates considering the zoom. This avoids to have, for
+            // example, a move of 4 pixels when the zoom level is 25%
+            double zoomFactor = GraphicalHelper.getZoom(getSourceEditPart());
+            PrecisionPoint preciseDelta = (PrecisionPoint) new PrecisionPoint(moveDelta).scale(zoomFactor);
+            // Set the request according to the move
+            ChangeBoundsRequest req = (ChangeBoundsRequest) getTargetRequest();
+            if (req.getMoveDelta().preciseX() == 0 && req.getMoveDelta().preciseY() == 0) {
+                req.setMoveDelta(preciseDelta);
+            } else {
+                req.setMoveDelta(req.getMoveDelta().getTranslated(preciseDelta));
+            }
```

Comment:
Yes because by default, the `getMoveDelta` returns a `Point` and not a `PrecisionPoint`. In this case, the result of `.getTranslated` ignore a precisionDelta of `0.5` for example.
So for the first time, when x and y = 0, we directly set with the preciseDelta.
No change will be done for this remark in the next PR version.